### PR TITLE
feat: improved test fixtures

### PR DESCRIPTION
**Pull request by roeap**
# Description

This PR is mainly preparation for the work we need to do on data skipping.
It intorduces a new set of test utilities, but at least also gets rid of one other "test_utils" file. Over the next few PRs I hope to gradually consolidate our usage on "Localized" test utitlitis and use the shared ones.

Initially I wanted to place this in the `test` crate, but somehow cargo kept on complaining about version conflicts, regarding the action structs.

We also add a function to get the partition values schema on snapshot, which I forgot to remove in rebasing some commits. WE will need it going forward so I was hoping it does not add too much complexity to this PR.

### CODE REVIEW

File: crates/core/src/test_utils/factories/mod.rs
Comment (ion-elgreco) [inline crates/core/src/test_utils/factories/mod.rs:1]: Nice!

File: crates/core/src/test_utils/factories/actions.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    pub fn into_action(self) -> Action {
+        Action::Protocol(self)
+    }
+}
+
+impl Metadata {
+    pub fn into_action(self) -> Action {
+        Action::Metadata(self)
+    }
+}
```
Comment (ion-elgreco) [inline crates/core/src/test_utils/factories/actions.rs:138]: We already have the `From<Add/Remove/Protocol/Metadata> for Action` in src/kernel/models/mod.rs, why is this required?

File: crates/core/src/operations/transaction/state.rs [hunk 2]
```diff
@@ -385,6 +412,6 @@ mod tests {
             .unwrap()
             .collect::<Vec<_>>();
         assert_eq!(files.len(), 2);
-        assert!(files.iter().all(|add| add.path.contains("included")));
+        // assert!(files.iter().all(|add| add.path.contains("included")));
```
Comment (ion-elgreco) [inline crates/core/src/operations/transaction/state.rs:415]: Can be removed, right? Since the add.paths are now random UUIDs

File: crates/core/src/test_utils/factories/actions.rs (hunk 1 above)
Comment (roeap) [reply crates/core/src/test_utils/factories/actions.rs]: it is not, also, now its gone :).

**Review by ion-elgreco — APPROVED** [review]
Great work!
